File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class Main {

    static List<Integer> dfs = new ArrayList<>();
    static List<Integer> bfs = new ArrayList<>();
    static int[][] graph;
    static int n;
    static int m;
    static int start;

    static boolean[] visited;

    public static void dfs(int number) {
        dfs.add(number);
        visited[number] = true;

        for (int i = 1; i <= n; i++) {
            if (graph[number][i] == 1 && !visited[i]) {
                dfs(i);
            }
        }
    }

    public static void bfs(int number) {
        Queue<Integer> q = new LinkedList<>();
        q.add(number);
        visited[number] = true;

        while (!q.isEmpty()) {
            int node = q.poll();
            bfs.add(node);

            for (int nv = 1; nv <= n; nv++) {
                if (graph[node][nv] == 1 && !visited[nv]) {
                    visited[nv] = true;
                    q.add(nv);
                }
            }
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        n = Integer.parseInt(input[0]);
        m = Integer.parseInt(input[1]);
        start = Integer.parseInt(input[2]);

        graph = new int[n + 1][n + 1];
        visited = new boolean[n + 1];

        for (int i = 0; i < m; i++) {
            String[] input2 = br.readLine().split(" ");
            int first = Integer.parseInt(input2[0]);
            int second = Integer.parseInt(input2[1]);
            graph[first][second] = 1;
            graph[second][first] = 1;
        }

        dfs(start);
        Arrays.fill(visited, false);
        bfs(start);

        System.out.println(dfs.toString()
                .replace(",","")
                .replace("[","")
                .replace("]",""));


        System.out.println(bfs.toString()
                .replace(",","")
                .replace("[","")
                .replace("]",""));

    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
from collections import deque
import sys


def dfs(number):
    dfs_result.append(number)
    visited[number] = True

    for i in range(1, n + 1):
        if graph[number][i] == 1 and not visited[i]:
            dfs(i)


def bfs(number):
    q = deque()
    q.append(number)
    visited[number] = True

    while q:
        node = q.popleft()
        bfs_result.append(node)

        for nv in range(1, n + 1):
            if graph[node][nv] == 1 and not visited[nv]:
                visited[nv] = True
                q.append(nv)


# 입력 처리
n, m, start = map(int, sys.stdin.readline().strip().split())

graph = [[0] * (n + 1) for _ in range(n + 1)]
visited = [False] * (n + 1)

for _ in range(m):
    first, second = map(int, sys.stdin.readline().strip().split())
    graph[first][second] = 1
    graph[second][first] = 1

dfs_result = []
bfs_result = []

dfs(start)
visited = [False] * (n + 1)
bfs(start)

print(' '.join(map(str, dfs_result)))
print(' '.join(map(str, bfs_result)))
